src/components/ControlPanel.jsx

```diff
@@ -1,19 +1,57 @@
-import React, { useContext } from 'react';
+/* eslint-disable no-debugger */
+import React, { useContext, useState } from 'react';
 import styled from 'styled-components';
 import { MoneyContext } from 'components/App';
+import ModifiableInput from 'components/ModifiableInput';
 
 function ControlPanel() {
   const { curMoney } = useContext(MoneyContext);
+  const [inputValue, setInputValue] = useState(curMoney);
+  const [isInputMode, setInputMode] = useState(false);
+  const UnmodifiableInput = React.useCallback(
+    () => getUnmodifiableInput({ value: curMoney, handler: handleInputMode }),
+    [],
```

Comment:
```suggestion
    [curMoney, handleInputMode],
```

- 의존성 배열에 콜백이 참조하는 값들을 모두 명시해주세요.
(참고: https://ko.reactjs.org/docs/hooks-reference.html#usecallback)
- 여기서 useCallback을 사용하신 이유가 무엇인지 궁금하네요!

Reply:
**useCallback**을 사용하기 이전에는 다음과 같이 컴포넌트가 중첩된 구조를 가졌습니다.

```js
function ControlPanel() {
  function ModifiableInput() {
    return <div />;
  }
  function UnmodifiableInput() {
    return <div />;
  }

  return (
    <div>
      {isInputMode ? <ModifiableInput /> : <UnmodifiableInput />}
    </div>
  );
}
```

그래서 [react/no-unstable-nested-components](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-unstable-nested-components.md) 관련 문서를 찾아보니

1. 컴포넌트를 밖으로 분리
2. **useCallback**을 활용해 메모이제이션해서 사용

을 통해서 중첩된 구조에 대한 문제를 해결할 수 있다고 이해했고, 그 중 2번을 선택해서 해결해보고 싶었습니다.

원래 `<ModifiableInput />` 컴포넌트도 **useCallback**을 활용하고 싶었지만 상태가 변할때마다 focus가 사라지는 이슈때문에 컴포넌트를 밖으로 분리했습니다.

[focus 사라지는 이슈 관련 영상](https://west-train-b77.notion.site/2022-05-17-16f57a38e2cc4653ad8caae892076141)입니다. (스크롤 조금만 내려주시면 관련 영상이 있습니다.)

2번 이유를 선택했던 이유는 `<ControlPanel />` 에서만 `<ModifiableInput />`, `<UnModifiableInput />` 를 사용하기 때문에 해당 컴포넌트 내부에서 정의하고 싶었습니다.

말씀해주셔서 다시 보니 코드의 일관성이 떨어져 보이는 것 같습니다.

Reply:
`useCallback` 으로 함수형 컴포넌트를 메모이제이션하려는 건 안티패턴으로 보여요. 또한 `useCallback` 은 함수를 메모이제이션하기때문에 **함수의 재생성을 막을 뿐이지, 함수의 결과값을 메모이제이션하지 않아서** 메모이제이션의 의미도 크지 않아 보입니다.

만들어주신 `getUnmodifiableInput` 함수는 결국 리액트 함수형 컴포넌트로 치환할 수 있어요. 컴포넌트(정확히는 함수형 컴포넌트의 결과값)를 메모이제이션하고 싶으신거라면, `React.memo` 를 한 번 공부해보시면 좋을 거 같습니다!